test/e2e/e2e_test.go

```diff
@@ -0,0 +1,251 @@
+	}
+
+	runCommand(t, webhookDeployPath, "--kubectl", kubectlBinaryPath, "--mode", "orchestrated")
+	checkPodMutation(t, "fpga.intel.com/region-ce48969398f05f33946d560708be108a")
+}
+
+func checkPodMutation(t *testing.T, expectedMutation string) {
+	podYamlFileName, err := getTempPodYaml()
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+	defer os.RemoveAll(podYamlFileName)
+
+	runCommand(t, kubectlBinaryPath, "wait", "--for=condition=Available", "-A", "-l", "app=intel-fpga-webhook", "--timeout=30s", "deployment")
+	runCommand(t, kubectlBinaryPath, "apply", "-f", podYamlFileName)
+
+	var success bool
+	for i := 0; i < 10; i++ {
+		time.Sleep(1 * time.Second)
+		cmd := exec.Command(kubectlBinaryPath, "get", "pod", "sample-pod", "-o", "jsonpath='{.spec.containers[0].resources.limits}'")
+		cmd.Env = append(os.Environ(), "KUBECONFIG="+kubeConfigPath)
+		output, err := cmd.Output()
+		if err != nil {
+			t.Fatalf("unable to check sample pod: %+v", err)
+		}
+		fmt.Println(string(output))
+		if bytes.Contains(output, []byte(expectedMutation)) {
+			success = true
+			break
+		}
+	}
+	if !success {
+		t.Error("sample pod hasn't been mutated")
+	}
+}
+
+func getTempPodYaml() (string, error) {
+	podYamlFile, err := ioutil.TempFile(cacheDir, "sample-pod")
+	if err != nil {
+		return "", errors.Wrap(err, "unable to create Pod config file")
+	}
+
+	if _, err := podYamlFile.Write([]byte(samplePod)); err != nil {
+		return "", errors.Wrap(err, "unable to write Pod config file")
+	}
+
+	if err := podYamlFile.Close(); err != nil {
+		return "", errors.Wrap(err, "unable to close Pod config file")
+	}
+
+	return podYamlFile.Name(), nil
+}
+
+func runCommand(t *testing.T, name string, arg ...string) {
+	cmd := exec.Command(name, arg...)
+	cmd.Env = append(os.Environ(), "KUBECONFIG="+kubeConfigPath)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		t.Fatalf("failed to run %q: %+v", name+" "+strings.Join(arg, " "), err)
+	}
+}
+
+func locateWebhookDeployerScript() (string, error) {
+	root := os.Getenv("PLUGINS_REPO_DIR")
+	if root != "" {
+		path := filepath.Join(root, "scripts/webhook-deploy.sh")
+		if _, err := os.Stat(path); !os.IsNotExist(err) {
+			return path, nil
+		}
+	}
+
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	path := filepath.Join(currentDir, "scripts/webhook-deploy.sh")
+	if _, err := os.Stat(path); !os.IsNotExist(err) {
+		return path, nil
+	}
+	path = filepath.Join(currentDir, "../../scripts/webhook-deploy.sh")
+	if _, err := os.Stat(path); !os.IsNotExist(err) {
+		return path, err
+	}
+
+	return "", errors.New("no script found, try to define PLUGINS_REPO_DIR pointing to the root of the repository")
+}
+
+func fetchExecutable(path, url string) error {
+	f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE, 0770)
+	if err != nil {
+		return errors.Wrapf(err, "unable to open file %q", path)
+	}
+	defer f.Close()
+	if err := downloadFromURL(url, f); err != nil {
```

Comment:
Have I understood correctly that this downloading happens for the same files for every test case?

Reply:
First the code checks if the cache exists. And only if it doesn't exist fetches the binaries.